tests/jballerina-unit-test/src/test/resources/test-src/workers/workers.bal

```diff
@@ -543,6 +543,7 @@ function waitFor() {
 }
 
 function add2(int i, int j) returns int {
+   runtime:sleep(1000);
```

Comment:
why we have to add a sleep here? do we know the root cause of this test failure ?

Reply:
It is failing because all three code blocks are running parallely. At one point w2 can start and go to sleep then the future can start and run to completion in 1s and then w2 can resume and the result is returned before w1 can run. Adding a sleep will delay the future also so w1 will for sure run to completion before w2 and the add2 method.
I think the test is to test the scenario of `First cancel the future and then wait `. So this change will make sure the orders are correct. `cancel the future and wait`

Reply:
after this sleep you have intriduced, can you try and run the this partcular test in a loop and verify?

Reply:
I have verified it by running it a 1000 times